compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/StreamingCodeDesugar.java

```diff
@@ -0,0 +1,812 @@
+        simpleSelectLambdaFunction.function = simpleSelectLambdaFunctionNode;
+        BLangBlockStmt lambdaBody = ASTBuilderUtil.createBlockStmt(selectClause.pos);
+        simpleSelectLambdaFunctionNode.requiredParams.add(lambdaFunctionVariable);
+        simpleSelectLambdaFunctionNode.returnTypeNode = ASTBuilderUtil.createTypeNode(symTable.anyType);
+        BLangValueType returnType = new BLangValueType();
+        returnType.setTypeKind(TypeKind.ANY);
+        simpleSelectLambdaFunctionNode.setReturnTypeNode(returnType);
+        defineFunction(simpleSelectLambdaFunctionNode, env.enclPkg);
+
+        simpleSelectLambdaFunctionNode.body = lambdaBody;
+        simpleSelectLambdaFunctionNode.closureVarSymbols.add(lambdaFunctionVariable.symbol);
+        simpleSelectLambdaFunctionNode.desugared = false;
+        simpleSelectLambdaFunction.pos = selectClause.pos;
+        simpleSelectLambdaFunction.type = simpleSelectLambdaFunctionNode.symbol.type;
+
+        //Create type casting for the filter variable
+        BVarSymbol typeCastingVarSymbol = new BVarSymbol(0,
+                new Name(getVariableName(lambdaFunctionVariable.getName().getValue())),
+                lambdaFunctionVariable.symbol.pkgID, inputStreamEventType, env.scope.owner);
+
+        BLangFieldBasedAccess eventObjectField = (BLangFieldBasedAccess) TreeBuilder.createFieldBasedAccessNode();
+        eventObjectField.expr = ASTBuilderUtil.createVariableRef(selectClause.pos, lambdaFunctionVariable.symbol);
+        eventObjectField.type = symTable.anyType;
+        eventObjectField.symbol = ((BRecordType) (lambdaFunctionVariable).type).fields.get(1).symbol;
+        eventObjectField.fieldKind = FieldKind.SINGLE;
+        eventObjectField.pos = selectClause.pos;
+        eventObjectField.field = ASTBuilderUtil.createIdentifier(selectClause.pos, "eventObject");
+
+        BLangExpression typeCastingExpression = generateConversionExpr(eventObjectField, inputStreamEventType,
+                symResolver);
+
+        BLangVariable typeCastingVariable = ASTBuilderUtil.
+                createVariable(selectClause.pos, getVariableName(lambdaFunctionVariable.getName().getValue()),
+                        inputStreamEventType, typeCastingExpression, typeCastingVarSymbol);
+
+        typeCastingVariable.typeNode = ASTBuilderUtil.createTypeNode(inputStreamEventType);
+        BLangVariableDef typeCastingVariableDef = ASTBuilderUtil.createVariableDef(selectClause.pos,
+                typeCastingVariable);
+        lambdaBody.stmts.add(typeCastingVariableDef);
+
+        //Output object creation
+        BLangRecordLiteral outputEventRecordLiteral = ASTBuilderUtil.createEmptyRecordLiteral(selectClause.pos,
+                outputEventTypeVariable.type);
+        List<BLangRecordLiteral.BLangRecordKeyValue> recordKeyValueList = new ArrayList<>();
+
+        List<? extends SelectExpressionNode> selectExprList = selectClause.getSelectExpressions();
```

Comment:
Since there are some complex desugaring happening here, shall we add some comments mentioning what each block does?

Reply:
Yes, done.